plugins/destination/s3/client/client_test.go

```diff
@@ -125,6 +127,8 @@ func testPluginCustom(t *testing.T, s *spec.Spec) {
 		t.Fatal(fmt.Errorf("failed to close client: %w", err))
 	}
 
+	time.Sleep(2 * time.Second)
```

Comment:
can we signal completion instead of waiting an arbitrary amount of time?

Reply:
But it has already been completed, code runs fine without the sleep if you're lucky (if S3 behaves well). This is to "unflake" the test. We could also mock S3 here, but that would probably go outside of the scope of this PR, which is to get the StreamingBatchWriter updates safely over the line.

Reply:
Got it! Totally makes sense to keep mocking out of scope of this PR.

Perhaps, something like [assert.Eventually](https://pkg.go.dev/github.com/stretchr/testify/assert#Eventually) could be a better fit here? Feel free to disregard the suggestion and proceed with the merge, though

Reply:
Good tip, implemented in all three PRs. Can't do it in the file PR unfortunately, without changing the test concept.